lib/publiccloud/azure.pm

```diff
@@ -283,10 +283,21 @@ sub parse_instance_id
 }
 
 =head2 cleanup
-This method is called called after each test on failure or success to revoke the credentials
+This method is called after each test module to cleanup
 =cut
 
 sub cleanup {
+    my ($self) = @_;
+    $self->SUPER::cleanup();
+    $self->provider_client->cleanup();
```

Comment:
and if you will follow my suggestion about  dropping `$self->provider_client->cleanup();`  you may also drop this implementation of "cleanup" because overloading method in child class just to call parent implementation does not make any sense